### ISSUE TITLE
Postgres Dialect: PRS error: `CREATE SERVER` `CREATE USER MAPPING FOR <db>` `IMPORT FOREIGN SCHEMA`

### ISSUE THREAD

**Issue by zenput-paul** [sqlfluff/sqlfluff#3927]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Parsing of `CREATE SERVER` `CREATE USER MAPPING FOR <db>` `IMPORT FOREIGN SCHEMA` failed

### Expected Behaviour

Queries are parsed correctly

### Observed Behaviour

```
[L:  1, P:  1]      |file:
[L:  1, P:  1]      |    statement:
[L:  1, P:  1]      |        unparsable:                                           !! Expected: 'statement'
[L:  1, P:  1]      |            raw:                                              'CREATE'
[L:  1, P:  7]      |            whitespace:                                       ' '
[L:  1, P:  8]      |            raw:                                              'SERVER'
[L:  1, P: 14]      |            whitespace:                                       ' '
[L:  1, P: 15]      |            raw:                                              'my_server'
[L:  1, P: 24]      |            newline:                                          '\n'
[L:  2, P:  1]      |            raw:                                              'FOREIGN'
[L:  2, P:  8]      |            whitespace:                                       ' '
[L:  2, P:  9]      |            raw:                                              'DATA'
[L:  2, P: 13]      |            whitespace:                                       ' '
[L:  2, P: 14]      |            raw:                                              'WRAPPER'
[L:  2, P: 21]      |            whitespace:                                       ' '
[L:  2, P: 22]      |            raw:                                              'mysql_fdw'
[L:  2, P: 31]      |            newline:                                          '\n'
[L:  3, P:  1]      |            raw:                                              'OPTIONS'
[L:  3, P:  8]      |            whitespace:                                       ' '
[L:  3, P:  9]      |            bracketed:
[L:  3, P:  9]      |                start_bracket:                                '('
[L:  3, P: 10]      |                raw:                                          'host'
[L:  3, P: 14]      |                whitespace:                                   ' '
[L:  3, P: 15]      |                single_quote:                                 "'mysql'"
[L:  3, P: 22]      |                comma:                                        ','
[L:  3, P: 23]      |                whitespace:                                   ' '
[L:  3, P: 24]      |                raw:                                          'port'
[L:  3, P: 28]      |                whitespace:                                   ' '
[L:  3, P: 29]      |                single_quote:                                 "'3306'"
[L:  3, P: 35]      |                end_bracket:                                  ')'
[L:  3, P: 36]      |    statement_terminator:                                     ';'
[L:  3, P: 37]      |    newline:                                                  '\n'
[L:  4, P:  1]      |    newline:                                                  '\n'
[L:  5, P:  1]      |    statement:
[L:  5, P:  1]      |        create_user_statement:
[L:  5, P:  1]      |            keyword:                                          'CREATE'
[L:  5, P:  7]      |            whitespace:                                       ' '
[L:  5, P:  8]      |            keyword:                                          'USER'
[L:  5, P: 12]      |            whitespace:                                       ' '
[L:  5, P: 13]      |            role_reference:
[L:  5, P: 13]      |                naked_identifier:                             'MAPPING'
[L:  5, P: 20]      |        unparsable:                                           !! Expected: 'Nothing...'
[L:  5, P: 20]      |            whitespace:                                       ' '
[L:  5, P: 21]      |            raw:                                              'FOR'
[L:  5, P: 24]      |            whitespace:                                       ' '
[L:  5, P: 25]      |            raw:                                              'my_db'
[L:  5, P: 30]      |            whitespace:                                       ' '
[L:  5, P: 31]      |            raw:                                              'SERVER'
[L:  5, P: 37]      |            whitespace:                                       ' '
[L:  5, P: 38]      |            raw:                                              'my_server'
[L:  5, P: 47]      |            whitespace:                                       ' '
[L:  5, P: 48]      |            raw:                                              'OPTIONS'
[L:  5, P: 55]      |            whitespace:                                       ' '
[L:  5, P: 56]      |            bracketed:
[L:  5, P: 56]      |                start_bracket:                                '('
[L:  5, P: 57]      |                raw:                                          'username'
[L:  5, P: 65]      |                whitespace:                                   ' '
[L:  5, P: 66]      |                single_quote:                                 "'root'"
[L:  5, P: 72]      |                end_bracket:                                  ')'
[L:  5, P: 73]      |    statement_terminator:                                     ';'
[L:  5, P: 74]      |    newline:                                                  '\n'
[L:  6, P:  1]      |    newline:                                                  '\n'
[L:  7, P:  1]      |    statement:
[L:  7, P:  1]      |        unparsable:                                           !! Expected: 'statement'
[L:  7, P:  1]      |            raw:                                              'IMPORT'
[L:  7, P:  7]      |            whitespace:                                       ' '
[L:  7, P:  8]      |            raw:                                              'FOREIGN'
[L:  7, P: 15]      |            whitespace:                                       ' '
[L:  7, P: 16]      |            raw:                                              'SCHEMA'
[L:  7, P: 22]      |            whitespace:                                       ' '
[L:  7, P: 23]      |            raw:                                              'my_db'
[L:  7, P: 28]      |            newline:                                          '\n'
[L:  8, P:  1]      |            raw:                                              'LIMIT'
[L:  8, P:  6]      |            whitespace:                                       ' '
[L:  8, P:  7]      |            raw:                                              'TO'
[L:  8, P:  9]      |            whitespace:                                       ' '
[L:  8, P: 10]      |            bracketed:
[L:  8, P: 10]      |                start_bracket:                                '('
[L:  8, P: 11]      |                newline:                                      '\n'
[L:  9, P:  1]      |                whitespace:                                   '    '
[L:  9, P:  5]      |                raw:                                          'table_a'
[L:  9, P: 12]      |                comma:                                        ','
[L:  9, P: 13]      |                newline:                                      '\n'
[L: 10, P:  1]      |                whitespace:                                   '    '
[L: 10, P:  5]      |                raw:                                          'table_b'
[L: 10, P: 12]      |                comma:                                        ','
[L: 10, P: 13]      |                newline:                                      '\n'
[L: 11, P:  1]      |                whitespace:                                   '    '
[L: 11, P:  5]      |                raw:                                          'table_c'
[L: 11, P: 12]      |                newline:                                      '\n'
[L: 12, P:  1]      |                end_bracket:                                  ')'
[L: 12, P:  2]      |            newline:                                          '\n'
[L: 13, P:  1]      |            raw:                                              'FROM'
[L: 13, P:  5]      |            whitespace:                                       ' '
[L: 13, P:  6]      |            raw:                                              'SERVER'
[L: 13, P: 12]      |            whitespace:                                       ' '
[L: 13, P: 13]      |            raw:                                              'my_server'
[L: 13, P: 22]      |            whitespace:                                       ' '
[L: 13, P: 23]      |            raw:                                              'INTO'
[L: 13, P: 27]      |            whitespace:                                       ' '
[L: 13, P: 28]      |            raw:                                              'public'
[L: 13, P: 34]      |    statement_terminator:                                     ';'
[L: 13, P: 35]      |    newline:                                                  '\n'
[L: 14, P:  1]      |    [META] end_of_file:

==== parsing violations ====
L:   1 | P:   1 |  PRS | Line 1, Position 1: Found unparsable section: 'CREATE SERVER
                       | my_server\nFOREIGN DATA WRA...'
L:   5 | P:  20 |  PRS | Line 5, Position 20: Found unparsable section: ' FOR my_db SERVER
                       | my_server OPTIONS (use...'
L:   7 | P:   1 |  PRS | Line 7, Position 1: Found unparsable section: 'IMPORT FOREIGN
                       | SCHEMA my_db\nLIMIT TO (\n ...'
WARNING: Parsing errors found and dialect is set to 'postgres'. Have you configured your dialect correctly?
```

### How to reproduce

1. Create a test.sql file with the next code 
https://www.postgresql.org/docs/current/sql-createserver.html
https://www.postgresql.org/docs/14/sql-createusermapping.html
https://www.postgresql.org/docs/14/sql-importforeignschema.html
```SQL
CREATE SERVER my_server
FOREIGN DATA WRAPPER mysql_fdw
OPTIONS (host 'mysql', port '3306');

CREATE USER MAPPING FOR my_db SERVER my_server OPTIONS (username 'root');

IMPORT FOREIGN SCHEMA my_db
LIMIT TO (
    table_a,
    table_b,
    table_c
)
FROM SERVER my_server INTO public;
```
2. Run linting command
```
sqlfluff parse --dialect postgres test.sql
```

### Dialect

Postgres

### Version

sqlfluff, version 1.3.1

### Configuration

```
[sqlfluff]
templater = jinja
sql_file_exts = .sql

[sqlfluff:indentation]
indented_joins = false
indented_using_on = true
template_blocks_indent = false

[sqlfluff:templater]
unwrap_wrapped_queries = true

[sqlfluff:templater:jinja]
apply_dbt_builtins = true
library_path = sqlfluff_mock_libs

[sqlfluff:rules:L016]
# Keywords
ignore_comment_lines = True
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
